linera-views/src/backends/scylla_db.rs

```diff
@@ -743,21 +759,32 @@ impl AdminKeyValueStore for ScyllaDbStoreInternal {
             .max_concurrent_queries
             .map(|n| Arc::new(Semaphore::new(n)));
         let max_stream_queries = config.max_stream_queries;
-        let root_key = get_big_root_key(&[]);
         Ok(Self {
             store,
             semaphore,
             max_stream_queries,
+        })
+    }
+
+    fn open_exclusive(&self, root_key: &[u8]) -> Result<Self::Store, ScyllaDbStoreInternalError> {
+        let store = self.store.clone();
+        let semaphore = self.semaphore.clone();
+        let max_stream_queries = self.max_stream_queries;
+        let root_key = get_big_root_key(root_key);
+        Ok(ScyllaDbStoreInternal {
+            store,
+            semaphore,
+            max_stream_queries,
             root_key,
         })
     }
 
-    fn open_exclusive(&self, root_key: &[u8]) -> Result<Self, ScyllaDbStoreInternalError> {
+    fn open_shared(&self, root_key: &[u8]) -> Result<Self::Store, ScyllaDbStoreInternalError> {
```

Comment:
Here the implementation of `open_shared` is the same as `open_exclusive`.
That's a little unfortunate to have code paste.

Reply:
yes we can change that